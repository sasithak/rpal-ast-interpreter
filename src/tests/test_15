let 
.within
..and
...rec
....function_form
.....<ID:reduceStart>
.....<ID:s>
.....<ID:n>
.....->
......eq
.......<ID:n>
.......<INT:0>
......<ID:s>
......gamma
.......gamma
........<ID:reduceStart>
........gamma
.........<ID:Stern>
.........<ID:s>
.......-
........<ID:n>
........<INT:1> 
...rec
....function_form
.....<ID:reduceEnd>
.....<ID:s>
.....<ID:m>
.....->
......eq
.......<ID:m>
.......<INT:0>
......<STR:''>
......gamma
.......gamma
........<ID:Conc>
........gamma
.........<ID:Stem>
.........<ID:s>
.......gamma 
........gamma
.........<ID:reduceEnd>
.........gamma
..........<ID:Stern>
..........<ID:s>
........-
.........<ID:m>
.........<INT:1>
...rec
....function_form
.....<ID:length>
.....<ID:s>
.....->
......eq
.......<ID:s>
.......<STR:''>
......<INT:0>
......+
.......gamma
........<ID:length>
........gamma
.........<ID:Stern>
.........<ID:s>
.......<INT:1>
..function_form
...<ID:substring> 
...<ID:s>
...<ID:n>
...<ID:m>
...->
....not
.....gamma
......<ID:Isstring>
......<ID:s>
....<STR:'s should be a string'>
....->
.....not
......gamma
.......<ID:Isinteger>
.......<ID:n>
.....<STR:'n should be an integer'>
.....->
......not
.......gamma
........<ID:Isinteger>
........<ID:m>
......<STR:'m should be an integer'>
......->
.......ls
........<ID:n>
........<INT:0>
.......<STR:'n should be non-negative'>
.......->
........ls
.........<ID:m>
.........<INT:0> 
........<STR:'m should be non-negative'>
........->
.........gr
..........+
...........<ID:n>
...........<ID:m>
..........gamma
...........<ID:length>
...........<ID:s>
.........<STR:'Index out of range'>
.........gamma
..........gamma
...........<ID:reduceEnd>
...........gamma
............gamma
.............<ID:reduceStart>
.............<ID:s>
............<ID:n>
..........<ID:m>
.gamma
..<ID:Print>
..tau
...gamma
....gamma
.....gamma
......<ID:substring>
......<STR:'abcdefghijkl'>
.....<INT:3>
....<INT:5>
...gamma
....gamma
.....gamma 
......<ID:substring>
......<STR:'abcd'>
.....<INT:3>
....<INT:7>
...gamma
....gamma
.....gamma
......<ID:substring>
......<STR:'abcd'>
.....<INT:5>
....<INT:8>
...gamma
....gamma
.....gamma
......<ID:substring>
......<STR:'abcd'>
.....<STR:'a'>
....<INT:2>
...gamma
....gamma
.....gamma
......<ID:substring>
......<STR:'abcd'>
.....<INT:2>
....<STR:'a'> 
...gamma
....gamma
.....gamma
......<ID:substring>
......<STR:'abcd'>
.....neg
......<INT:2>
....<INT:5>
...gamma
....gamma
.....gamma
......<ID:substring>
......<STR:'abcd'>
.....<INT:2>
....neg
.....<INT:5>
