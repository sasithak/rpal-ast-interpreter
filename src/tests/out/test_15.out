(defgh, Index out of range, Index out of range, n should be an integer, m should be an integer, n should be non-negative, m should be non-negative)
